src/shared/varlink.c

```diff
@@ -2263,11 +2274,24 @@ int varlink_server_listen_fd(VarlinkServer *s, int fd) {
                         return r;
         }
 
+        *ret_ss = TAKE_PTR(ss);
+        return 0;
+}
+
+int varlink_server_listen_fd(VarlinkServer *s, int fd) {
```

Comment:
Hmm, so shouldn't this take an optional `const char *address` to set the address? That way `varlink_server_listen_address()` would't need to duplicate the code.

Reply:
Combine it into something like `int varlink_server_listen_object(VarlinkServer *s, int fd, const char *address, mode_t m)` ? This would mean either fd or address is passed. That would reduce the duplicate lines, though IMO it is more readable the way it is now.